### PR TITLE
fix: failure domains reporting from microvmcluster

### LINKED ISSUE #14: Add `FailureDomain` support to cluster controller
**Describe the solution you'd like:**
Implement the `FailureDomain` support in the clusters controller where a fault domain refers to a flintlock bare-metal host

**Why do you want this feature:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:

The API definitions for the MicrovmMachine was incorrectly spcified
**FailureDomain** which is a deprecated way to specify a failure domain.
This has been removed, and the code now looks for failure domain on the
**Machine** instead as this is the correct way.

Declarative placement has been introduced at the **MicrovmCluster**
level with the ability to set a static list of host endpoints to use
when creating microvms. This relies on the failure domain logic to
assign a **Machine** to one of the host endpoints.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #14 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/cluster-template.yaml [hunk 1]
```diff
@@ -28,6 +28,11 @@ spec:
   controlPlaneEndpoint:
     host: "${CONTROL_PLANE_VIP}"
     port: 6443
+  placement:
+    staticPool:
+      hosts:
+      - endpoint: "${MICROVM_ENDPOINT:=127.0.0.1:9090}"
```
Comment (Callisto13) [inline templates/cluster-template.yaml]: Can we rename this var? It's a bit misleading

`HOST_ENDPOINT`, `MVM_SERVICE_ENDPOINT`, `FLINTLOCK_ADDRESS` or something which does not sound like it is the address to an actual mvm

(will need docs change)

File: controllers/microvmcluster_controller.go [hunk 2]
```diff
@@ -172,6 +176,28 @@ func (r *MicrovmClusterReconciler) isAPIServerAvailable(ctx context.Context, clu
...
+			}
+		}
+
+		clusterScope.MvmCluster.Status.FailureDomains = failureDomains
+
+		return nil
+	}
+	// NOTE: additional placement methods can be added the future
+
+	return errNoPlacement
```
Comment (Callisto13) [inline controllers/microvmcluster_controller.go]: can we switch around the default path here, so we don't do the actual work indented?

like
```suggestion
	if placement.StaticPool == nil {
	        return errNoPlacement
	}
	 
	 // actual stuff is default path
```

File: templates/cluster-template.yaml (hunk 1 above)
Comment (richardcase) [reply templates/cluster-template.yaml]: `HOST_ENDPOINT` sounds good. I will change...thanks.

File: controllers/microvmcluster_controller.go (hunk 2 above)
Comment (richardcase) [reply controllers/microvmcluster_controller.go]: Sure thing :)

File: templates/cluster-template.yaml (hunk 1 above)
Comment (richardcase) [reply templates/cluster-template.yaml]: Changed.

File: controllers/microvmcluster_controller.go (hunk 2 above)
Comment (richardcase) [reply controllers/microvmcluster_controller.go]: Changed.

**Review by Callisto13 — DISMISSED** [review]

**Review by yitsushi — DISMISSED** [review]

**Review by Callisto13 — APPROVED** [review]

**Pull request merged**